File: hackerrank/easy/0025-jumping-on-the-clouds-revisited/js/solution.mjs
// solution.mjs

export function jumpingOnClouds(c, k) {
  let energy = 100;
  let pos = 0;
  const n = c.length;
  do {
    pos = (pos + k) % n;
    energy -= c[pos] === 1 ? 3 : 1;
  } while (pos !== 0);
  return energy;
}

// --- demo run (from sample) ---
const c0 = [0, 0, 1, 0, 0, 1, 1, 0];
console.log(jumpingOnClouds(c0, 2)); // 92

File: hackerrank/easy/0025-jumping-on-the-clouds-revisited/swift/solution.swift
// solution.swift

func jumpingOnClouds(c: [Int], k: Int) -> Int {
    var energy = 100
    var pos = 0
    repeat {
        pos = (pos + k) % c.count
        energy -= (c[pos] == 1) ? 3 : 1
    } while pos != 0
    return energy
}

// --- demo run (from sample) ---
let c0 = [0, 0, 1, 0, 0, 1, 1, 0]
print(jumpingOnClouds(c: c0, k: 2)) // 92